src/components/DocExplorer/Argument.js

```diff
@@ -0,0 +1,46 @@
+/**
+ *  Copyright (c) Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the license found in the
+ *  LICENSE file in the root directory of this source tree.
+ */
+
+import React, { PropTypes } from 'react';
+import TypeLink from './TypeLink';
+
+export default class Argument extends React.Component {
+  static propTypes = {
+    arg: PropTypes.object.isRequired,
+    onClickType: PropTypes.func.isRequired,
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return this.props.arg !== nextProps.arg ||
```

Comment:
It doesn't check all the properties. I used this clause to treat only the `undefined` to `some_object` case, but taking a closer look it's wrong anyway.

I dig a bit more, and I think we don't even need the `shouldComponentUpdate` method defined in this case, I was trying to simulate it and it's never called.

What do you think?